pkg/query/storeset.go

```diff
@@ -349,6 +357,7 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 	s.storesStatusesMtx.Lock()
 	defer s.storesStatusesMtx.Unlock()
 
+	s.storeNodeStatus.WithLabelValues(store.addr).Set(0)
```

Comment:
Sorry, but I think we are making a mistake here. Why we are flipping gauge on every update? So e.g if some store API is up 100% time, prometheus can still see it being down because it scrapes concurrently to this code. 

So if scrape happens while we are in line 367 we share incorrect infomation. Please fix it by changing Gauge value here based on error all the time. Also not sure if this is a correct place, method name suggests we are updating `status` (for UI) NOT the metric. This might be quite confusing for a reader.

I belive setting this metric near the healthcheck is the best solution and the most intuitive. What do you think? (: